### DISCUSSION TITLE
v0.2.0

### DISCUSSION THREAD

**Discussion by meowgorithm** [charmbracelet/lipgloss#26] [General]
# Getters

This PR adds getters for all style rules, enabling you to query your style rules. This can be handy when building layouts, particularly when styles are variable, such as a component. For example:

```go
if headerStyle.GetMarginLeft() > lipgloss.Width(spinner) + lipgloss.Width(gap) {
    // jam the spinner in the left gutter
} else {
   // just put the spinner on the right
}
```

For details on the getter API, check out [the docs](https://pkg.go.dev/github.com/charmbracelet/lipgloss#section-documentation).

* * *

Thoughts? Questions? We love hearing from you. Feel free to reach out on [Twitter](https://twitter.com/charmcli), [The Fediverse](https://mastodon.technology/@charm) or right here in [GitHub Discussions](https://github.com/charmbracelet/lipgloss/discussions).

<a href="https://charm.sh/"><img alt="The Charm logo" src="https://stuff.charm.sh/charm-badge-unrounded.jpg" width="400"></a>

<hr /><em>This discussion was created from the release <a href='https://github.com/charmbracelet/lipgloss/releases/tag/v0.2.0'>v0.2.0</a>.</em>

**Comment by beltagyy** [comment]
v0.2.0 of Lipgloss brought significant refactoring, including improved color handling, new layout primitives, and better alignment helpers. If you are migrating from v0.1.x, the main breaking change is the switch to ColorProfile-aware rendering. Review the release notes for the full changelog and migration tips.
